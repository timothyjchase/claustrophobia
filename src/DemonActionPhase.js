import React from 'react'
import { observer } from 'mobx-react'
import { Button, Icon, Message } from 'semantic-ui-react'
import EventMessage from './EventMessage'

const DemonActionPhase = ({ game }) => (
  <div>
    {!!game.event &&
      game.event.phase === 'DEMON_ACTION' && <EventMessage game={game} />}
    {game.trogsSupernaturalSpeed && (
      <Message negative>
        <Message.Header>Supernatural Speed</Message.Header>
        <Message.Content>
          All Troglodytes gain <strong>+1 MVT</strong>.
        </Message.Content>
      </Message>
    )}
    {game.trogsSharpenedClaws && (
      <Message negative>
        <Message.Header>Sharpened Claws</Message.Header>
        <Message.Content>
          All Troglodytes are <strong>Frantic</strong>.
        </Message.Content>
      </Message>
    )}

    {game.scenario !== 'THE_RITUAL' && (
      <div>
        <p>
          <strong>Activation order</strong>: proximity to Human warriors
          (closest first).
        </p>
        <p>
          <strong>Action order</strong>: if starting activation on a tile with a
          Human warrior, attack then move. Otherwise, move then attack.
        </p>
        <p>
          <strong>Move</strong>: towards the closest Human warrior (largest
          group if tied).
        </p>
        <p>
          <strong>Attack</strong>: most wounded Human warrior (lowest defense if
          tied).
        </p>
      </div>
    )}

    {game.scenario === 'THE_RITUAL' && (
      <div>
        <p>
          <strong>Activation order</strong>: proximity to the Pentacle Room
          (closest first).
        </p>
        <p>
          <strong>Action order</strong>: if starting activation on a tile with a
          Human warrior, attack then move. Otherwise, move then attack.
        </p>
        <p>
          <strong>Move</strong>: Troglodytes move towards the Pentacle Room.
          {!!game.demonsInPlay && (
            <span>
              The <strong>Demon of Cruelty</strong> will not move.
            </span>
          )}
        </p>
        <p>
          <strong>Attack</strong>: most wounded Human warrior (lowest defense if
          tied).
        </p>
      </div>
    )}
    <br />
    <Button.Group vertical fluid>
      <Button primary onClick={() => game.completeDemonActionPhase()}>
        <Icon name="play" /> Next Phase
      </Button>
    </Button.Group>
  </div>
)

export default observer(DemonActionPhase)
