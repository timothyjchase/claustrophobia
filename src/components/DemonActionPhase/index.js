import React from 'react'
import PropTypes from 'prop-types'
import { Button, Message } from 'semantic-ui-react'

const DemonActionPhase = ({
  scenarioKey,
  trogsSupernaturalSpeed,
  trogsSharpenedClaws,
  demonsInPlay,
  completeDemonActionPhase,
}) => (
  <div>
    {trogsSupernaturalSpeed && (
      <Message negative>
        <Message.Header>Supernatural Speed</Message.Header>
        <Message.Content>
          <span>All Troglodytes gain </span>
          <strong>+1 MVT</strong>
          <span>.</span>
        </Message.Content>
      </Message>
    )}
    {trogsSharpenedClaws && (
      <Message negative>
        <Message.Header>Sharpened Claws</Message.Header>
        <Message.Content>
          <span>All Troglodytes are </span>
          <strong>Frantic</strong>
          <span>.</span>
        </Message.Content>
      </Message>
    )}

    {scenarioKey !== 'THE_RITUAL' && (
      <div>
        <p>
          <strong>Activation order: </strong>
          <span>proximity to Human warriors (closest first).</span>
        </p>
        <p>
          <strong>Action order: </strong>
          if on a tile with a Human warrior, attack then move. Otherwise, move
          then attack.
        </p>
        <p>
          <strong>Move: </strong>
          towards the closest Human warrior (largest group if tied).
        </p>
        <p>
          <strong>Attack: </strong>
          most wounded Human warrior (lowest defense if tied).
        </p>
      </div>
    )}

    {scenarioKey === 'THE_RITUAL' && (
      <div>
        <p>
          <strong>Activation order: </strong>
          proximity to the Pentacle Room (closest first).
        </p>
        <p>
          <strong>Action order: </strong>
          if on a tile with a Human warrior, attack then move. Otherwise, move
          then attack.
        </p>
        <p>
          <strong>Move: </strong>
          Troglodytes move towards the Pentacle Room.
          {!!demonsInPlay && (
            <span>
              <span>The </span>
              <strong>Demon of Cruelty </strong>
              will not move.
            </span>
          )}
        </p>
        <p>
          <strong>Attack: </strong>
          most wounded Human warrior (lowest defense if tied).
        </p>
      </div>
    )}
    <br />
    <Button.Group vertical fluid>
      <Button
        primary
        icon="play"
        content="Next Phase"
        onClick={completeDemonActionPhase}
      />
    </Button.Group>
  </div>
)

DemonActionPhase.defaultProps = {
  trogsSupernaturalSpeed: false,
  trogsSharpenedClaws: false,
}

DemonActionPhase.propTypes = {
  scenarioKey: PropTypes.string.isRequired,
  trogsSupernaturalSpeed: PropTypes.bool,
  trogsSharpenedClaws: PropTypes.bool,
  demonsInPlay: PropTypes.number.isRequired,
  completeDemonActionPhase: PropTypes.func.isRequired,
}

export default DemonActionPhase
