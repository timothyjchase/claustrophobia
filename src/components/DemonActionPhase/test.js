import React from 'react'
import DemonActionPhase from './index'

describe('<DemonActionPhase />', () => {
  const commonProps = {
    demonsInPlay: 0,
    completeDemonActionPhase: jest.fn(),
  }
  describe('with scenario = THE_POSSESSED', () => {
    const component = shallow(
      <DemonActionPhase {...commonProps} scenarioKey="THE_POSSESSED" />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with scenario = THE_POSSESSED, trogsSupernaturalSpeed = true, trogsSharpenedClaws = true', () => {
    const component = shallow(
      <DemonActionPhase
        {...commonProps}
        scenarioKey="THE_POSSESSED"
        trogsSupernaturalSpeed
        trogsSharpenedClaws
      />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with scenario = THE_RITUAL, demonsInPlay=0', () => {
    const component = shallow(
      <DemonActionPhase {...commonProps} scenarioKey="THE_RITUAL" />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with scenario = THE_RITUAL, demonsInPlay=1', () => {
    const component = shallow(
      <DemonActionPhase
        {...commonProps}
        scenarioKey="THE_RITUAL"
        demonsInPlay={1}
      />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
