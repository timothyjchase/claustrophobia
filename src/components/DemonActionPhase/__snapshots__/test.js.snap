// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DemonActionPhase /> with scenario = THE_POSSESSED should match snapshot 1`] = `
<div>
  <div>
    <p>
      <strong>
        Activation order: 
      </strong>
      <span>
        proximity to Human warriors (closest first).
      </span>
    </p>
    <p>
      <strong>
        Action order: 
      </strong>
      if on a tile with a Human warrior, attack then move. Otherwise, move then attack.
    </p>
    <p>
      <strong>
        Move: 
      </strong>
      towards the closest Human warrior (largest group if tied).
    </p>
    <p>
      <strong>
        Attack: 
      </strong>
      most wounded Human warrior (lowest defense if tied).
    </p>
  </div>
  <br />
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;

exports[`<DemonActionPhase /> with scenario = THE_POSSESSED, trogsSupernaturalSpeed = true, trogsSharpenedClaws = true should match snapshot 1`] = `
<div>
  <Message
    negative={true}
  >
    <MessageHeader>
      Supernatural Speed
    </MessageHeader>
    <MessageContent>
      <span>
        All Troglodytes gain 
      </span>
      <strong>
        +1 MVT
      </strong>
      <span>
        .
      </span>
    </MessageContent>
  </Message>
  <Message
    negative={true}
  >
    <MessageHeader>
      Sharpened Claws
    </MessageHeader>
    <MessageContent>
      <span>
        All Troglodytes are 
      </span>
      <strong>
        Frantic
      </strong>
      <span>
        .
      </span>
    </MessageContent>
  </Message>
  <div>
    <p>
      <strong>
        Activation order: 
      </strong>
      <span>
        proximity to Human warriors (closest first).
      </span>
    </p>
    <p>
      <strong>
        Action order: 
      </strong>
      if on a tile with a Human warrior, attack then move. Otherwise, move then attack.
    </p>
    <p>
      <strong>
        Move: 
      </strong>
      towards the closest Human warrior (largest group if tied).
    </p>
    <p>
      <strong>
        Attack: 
      </strong>
      most wounded Human warrior (lowest defense if tied).
    </p>
  </div>
  <br />
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;

exports[`<DemonActionPhase /> with scenario = THE_RITUAL, demonsInPlay=0 should match snapshot 1`] = `
<div>
  <div>
    <p>
      <strong>
        Activation order: 
      </strong>
      proximity to the Pentacle Room (closest first).
    </p>
    <p>
      <strong>
        Action order: 
      </strong>
      if on a tile with a Human warrior, attack then move. Otherwise, move then attack.
    </p>
    <p>
      <strong>
        Move: 
      </strong>
      Troglodytes move towards the Pentacle Room.
    </p>
    <p>
      <strong>
        Attack: 
      </strong>
      most wounded Human warrior (lowest defense if tied).
    </p>
  </div>
  <br />
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;

exports[`<DemonActionPhase /> with scenario = THE_RITUAL, demonsInPlay=1 should match snapshot 1`] = `
<div>
  <div>
    <p>
      <strong>
        Activation order: 
      </strong>
      proximity to the Pentacle Room (closest first).
    </p>
    <p>
      <strong>
        Action order: 
      </strong>
      if on a tile with a Human warrior, attack then move. Otherwise, move then attack.
    </p>
    <p>
      <strong>
        Move: 
      </strong>
      Troglodytes move towards the Pentacle Room.
      <span>
        <span>
          The 
        </span>
        <strong>
          Demon of Cruelty 
        </strong>
        will not move.
      </span>
    </p>
    <p>
      <strong>
        Attack: 
      </strong>
      most wounded Human warrior (lowest defense if tied).
    </p>
  </div>
  <br />
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;
